src/storage/orderbook.cpp

```diff
@@ -0,0 +1,103 @@
+#include "orderbook.hpp"
+
+namespace raccoon {
+namespace storage {
+
+void
+OrderbookProcessor::ob_to_redis_(redisContext* redis, const std::string& product_id)
+{
+    log_d(main, "Pushing orderbook {} to redis", product_id);
+
+    product_tracker tracker = orderbook_[product_id];
+    map_to_redis_(redis, tracker.asks, product_id + "-ASKS");
+    map_to_redis_(redis, tracker.bids, product_id + "-BIDS");
+}
+
+void
+OrderbookProcessor::process_incoming_update_(const OrderbookUpdate& newUpdate)
+{
+    log_d(main, "Processing incoming update for {}", newUpdate.product_id);
+
+    auto [it, inserted] = orderbook_.emplace(newUpdate.product_id, product_tracker());
+    product_tracker& tracker = it->second;
+
+    auto updateOrderbook =
+        [](std::unordered_map<double, double>& orderSide, double price, double volume) {
+            if (volume == 0.0) {
```

Comment:
```suggestion
            if (volume <= 0.0000000001) {
```
Otherwise GCC gets sad